library/azure_ad_users.py

```diff
@@ -0,0 +1,310 @@
+
+  mail_nick_name:
+    description:
+      - This is the Mail Nick Name for the user account.
+    required: false
+    default: If no argument is passed, it will be set to the UPN without the @Domain part.
+
+  given_name:
+    description:
+      - User account first name
+    required: false
+    default: null
+
+  surname:
+    description:
+      - User account last name.
+    required: false
+    default: null
+'''.format(__version__)
+
+EXAMPLES = '''
+# Basic user creation example
+tasks:
+- name: Create a new Azure user account
+  azure_ad_users:
+    user_name            : "ansible.test"
+    state           : present
+    password        : "Test1234"
+    tenant_domain    : "AnsibleDomain.onmicrosoft.com"
+    client_id       : "6359f1g62-6543-6789-124f-398763x98112"
+    client_secret   : "HhCDbhsjkuHGiNhe+RE4aQsdjjrdof8cSd/q8F/iEDhx="
+
+# Basic user deletion example
+tasks:
+- name: Delete one Azure user account
+  azure_ad_users:
+    user_name            : "ansible.test"
+    state           : absent
+    tenant_domain    : "AnsibleDomain.onmicrosoft.com"
+    client_id       : "6359f1g62-6543-6789-124f-398763x98112"
+    client_secret   : "HhCDbhsjkuHGiNhe+RE4aQsdjjrdof8cSd/q8F/iEDhx="
+
+'''
+
+class AzureAdUsers():
+    def __init__(self, module):
+        self.module = module
+        self.user_name = self.module.params["user_name"] + "@" + self.module.params["tenant_domain"]
+        self.state = self.module.params["state"]
+        self.password = self.module.params["password"]
+        self.change_password_onlogin = self.module.params["change_password_onlogin"]
+        self.enabled = self.module.params["enabled"]
+        self.display_name = self.module.params["display_name"]
+        if not self.display_name:
+            #i = self.user_name.split('@', 1)
+            self.display_name = self.module.params["user_name"]
+        self.mail_nick_name = self.module.params["mail_nick_name"]
+        if not self.mail_nick_name:
+            #i = self.user_name.split('@', 1)
+            self.mail_nick_name = self.module.params["user_name"]
+	self.immutable_id = self.module.params["immutable_id"]
+        self.given_name = self.module.params["given_name"]
+        self.surname = self.module.params["surname"]
+        self.tenant_domain = self.module.params["tenant_domain"]
+        self.client_id = self.module.params["client_id"]
+        self.client_secret = self.module.params["client_secret"]
+        self.graph_url = self.module.params["graph_url"]
+        self.login_url  = self.module.params["login_url"]
+        if not self.graph_url:
+            self.graph_url = "https://graph.windows.net/{}".format(self.tenant_domain)
+        if not self.login_url:
+            self.login_url = "https://login.windows.net/{}/oauth2/token?api-version=1.0".format(self.tenant_domain)
+
+        # Geting azure cred from ENV if not defined
+        if not self.client_id:
+            if 'azure_client_id' in os.environ:
```

Comment:
No need to check for different environment name. I would stick to the official azure environment name. As defined by Microsoft